mne/epochs.py

```diff
@@ -272,9 +272,10 @@ def __init__(self, info, data, events, event_id, tmin, tmax,
             self.preload = False
             self._data = None
         else:
+            index_offset = 1 if data.shape[-1] > 1 else 0
             assert decim == 1
             if data.ndim != 3 or data.shape[2] != \
-                    round((tmax - tmin) * self.info['sfreq']) + 1:
+                    round((tmax - tmin) * self.info['sfreq']) + index_offset:
```

Comment:
I agree, I hope we can avoid exceptional cases.


On Thu, Aug 27, 2015 at 7:54 PM, Denis-Alexander Engemann <
denis.engemann@gmail.com> wrote:

> I don't fully understand this. I can only tell you that it used to work
> before.
>
> On Thu, Aug 27, 2015 at 7:27 PM, Eric Larson <notifications@github.com>
> wrote:
>
>> In mne/epochs.py
>> <https://github.com/mne-tools/mne-python/pull/2435#discussion_r38122024>:
>>
>> >              assert decim == 1
>> >              if data.ndim != 3 or data.shape[2] != \
>> > -                    round((tmax - tmin) * self.info['sfreq']) + 1:
>> > +                    round((tmax - tmin) * self.info['sfreq']) + index_offset:
>>
>> In other words, I'm worried when the single-value case needs to be
>> exceptional. It really seems like it shouldn't need to be. The fact that we
>> are treating it as such might indicate a bigger problem with how we deal
>> with times, that we should fix at the root instead of implementing
>> workarounds. Maybe that's incorrect, though.
>>
>> —
>> Reply to this email directly or view it on GitHub
>> <https://github.com/mne-tools/mne-python/pull/2435/files#r38122024>.
>>
>
>